packages/server/graphql/mutations/addMissingJiraField.ts

```diff
@@ -0,0 +1,165 @@
+import MeetingPoker from '../../database/types/MeetingPoker'
+import EstimatePhase from '../../database/types/EstimatePhase'
+import getTemplateRefById from '../../postgres/queries/getTemplateRefById'
+import {TaskServiceEnum} from '../../database/types/Task'
+import JiraServiceTaskId from '~/shared/gqlIds/JiraServiceTaskId'
+import publish from '../../utils/publish'
+import {SubscriptionChannel} from '~/types/constEnums'
+import {JiraScreen, JiraScreensResponse, JiraScreenTab} from '~/utils/AtlassianManager'
+
+const addMissingJiraField = {
+  type: GraphQLNonNull(AddMissingJiraFieldPayload),
+  description: `Adds a missing Jira field to a screen currently assigned to a Jira project`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    },
+    stageId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (
+    _source,
+    {meetingId, stageId},
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    //AUTH
+    const meeting = (await dataLoader.get('newMeetings').load(meetingId)) as MeetingPoker
+    if (!meeting) {
+      return {error: {message: 'Meeting not found'}}
+    }
+    const {
+      endedAt,
+      phases,
+      meetingType,
+      teamId,
+      createdBy,
+      facilitatorUserId,
+      templateRefId
+    } = meeting
+    if (!isTeamMember(authToken, teamId)) {
+      return {error: {message: 'Not on the team'}}
+    }
+    if (endedAt) {
+      return {error: {message: 'Meeting has ended'}}
+    }
+    if (meetingType !== 'poker') {
+      return {error: {message: 'Not a poker meeting'}}
+    }
+    if (isPhaseComplete('ESTIMATE', phases)) {
+      return {error: {message: 'Estimate phase is already complete'}}
+    }
+    if (viewerId !== facilitatorUserId) {
+      if (viewerId !== createdBy) {
+        return {
+          error: {message: 'Not meeting facilitator'}
+        }
+      }
+      return {
+        error: {message: 'Not meeting facilitator anymore'}
+      }
+    }
+
+    // VALIDATION
+    const estimatePhase = phases.find((phase) => phase.phaseType === 'ESTIMATE')! as EstimatePhase
+    const {stages} = estimatePhase
+    const stage = stages.find((stage) => stage.id === stageId)
+    if (!stage) {
+      return {error: {message: 'Invalid stageId provided'}}
+    }
+
+    // RESOLUTION
+    const {creatorUserId, dimensionRefIdx, service, serviceTaskId} = stage
+    const templateRef = await getTemplateRefById(templateRefId)
+    const {dimensions} = templateRef
+    const dimensionRef = dimensions[dimensionRefIdx]
+    const {name: dimensionName} = dimensionRef
+    if ((service as TaskServiceEnum) !== 'jira') {
+      return {error: {message: 'Non Jira service'}}
+    }
+    const auth = await dataLoader.get('freshAtlassianAuth').load({teamId, userId: creatorUserId})
```

Comment:
I think it'd be enough to just use viewerId because just before calling this mutation a facilitator was asked to authorize with Jira, any thoughts @mattkrick?